.github/workflows/build_test.yml

```
@@ -0,0 +1,112 @@
+name: Standard Build and Test
+
+on:
+  # allows us to run workflows manually
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - develop
+      - master
+  push:
+    branches:
+      - develop
+      - master
+
+jobs:
+  Housekeeping:
+    runs-on: ubuntu-latest
+    container:
+      image: svalinn/dagmc-ci-ubuntu-18.04-housekeeping:latest
+    steps:
+      - name: Install latest git
+        run: |
+          apt-get install -y software-properties-common
+          add-apt-repository ppa:git-core/ppa
+          apt update
+          apt install -y git
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
```

This might be worth a try

```suggestion
      - name: Checkout repository
        uses: actions/checkout@v2
        with:
          submodules: 'true'```

---

already tried this does not work as git version is < 3.18

---

see here: https://github.com/svalinn/DAGMC/runs/3031336027?check_suite_focus=true

```
Run actions/checkout@v2
2  with:
3    submodules: recursive
4    repository: svalinn/DAGMC
5    token: ***
6    ssh-strict: true
7    persist-credentials: true
8    clean: true
9    fetch-depth: 1
10    lfs: false
11  env:
12    hdf5_versions: 1.10.4
13    hdf5_build_dir: hdf5_build_dir
14 /usr/bin/docker exec  47d233d87e73e379239b48b5ef537b79ae073afeb55c7f2006b218366d7d4468 sh -c "cat /etc/*release | grep ^ID"
15 Syncing repository: svalinn/DAGMC
16 Getting Git version info
20 Deleting the contents of '/__w/DAGMC/DAGMC'
21 The repository will be downloaded using the GitHub REST API
22 To create a local Git repository instead, add Git 2.18 or higher to the PATH
23 Error: Input 'submodules' not supported when falling back to download using the GitHub REST API. To create a local Git repository instead, add Git 2.18 or higher to the PATH.
```

---

Can we add the git update to the Dockerfiles, then?

---

yes I believe that's the way to go.

I added it there to make sure we all agreed on this solution :)

---

See #753